### PR TITLE
[DOC] Section to describe how to use a configmap for logging

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

- Documentation

### Description

New section that describes how a configmap can be used to define logging for a resource.
Procedure shows steps with examples.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: documentation/assemblies/assembly-external-logging.adoc [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// This assembly is included in the following assemblies:
+//
+// assembly-deployment-configuration.adoc
+
+[id='external-logging_{context}']
+= External logging
+
+When setting the logging levels for a resource, you can specify them _inline_ directly in the `spec` of the resource YAML:
```
Comment (ppatierno) [inline documentation/assemblies/assembly-external-logging.adoc]: in the `spec.logging` section of the resource? what do you think?

File: documentation/assemblies/assembly-external-logging.adoc [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+----
+spec:
+  # ...
+  logging:
+    type: external
+    name: _customConfigMap_
+----
+
+With external logging, logging properties are defined in a ConfigMap.
+The name of the ConfigMap is referenced in the `spec`.
```
Comment (ppatierno) [inline documentation/assemblies/assembly-external-logging.adoc]: do we want to be more precise saying `spec.logging.name` property/field?

File: documentation/modules/proc-creating-configmap.adoc [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+// assembly-external-logging.adoc
+
+[id='creating-configmap_{context}']
+= Creating a ConfigMap for logging
+
+To use a ConfigMap to define logging properties, you create the ConfigMap and then reference it as part of the logging definition in the `spec` of a resource.
+
+The ConfigMap must contain the appropriate logging configuration.
+
+* `log4j.properties` for Kafka components
```
Comment (ppatierno) [inline documentation/modules/proc-creating-configmap.adoc]: log4j is for the Kafka bridge as well.

File: documentation/modules/proc-creating-configmap.adoc [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+* `log4j2.properties` for the Topic Operator and User Operator
```
Comment (ppatierno) [inline documentation/modules/proc-creating-configmap.adoc]: should we say that the above `log4j.properties` and `log4j2.properties` are the fields in the ConfigMap data section where the file content should be put?

File: documentation/modules/proc-creating-configmap.adoc [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+[source,yaml,subs="+attributes"]
+----
+kind: ConfigMap
+apiVersion: {KafkaApiVersion}
+metadata:
+  name: logging-configmap
+data:
+  log4j.properties:
+  # logger settings for resource
+    kafka.root.logger: "INFO"
```
Comment (ppatierno) [inline documentation/modules/proc-creating-configmap.adoc]: the syntax in a properties file is with "=" not ":" so it should be 
kafka.root.logger=INFO

File: documentation/modules/proc-creating-configmap.adoc [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+----
+kubectl create configmap logging-configmap --from-file=log4j.properties
+----
++
+The properties file defines the logging configuration:
++
+[source,text]
+----
+# Define the root logger
+kafka.root.logger: "INFO"
```
Comment (ppatierno) [inline documentation/modules/proc-creating-configmap.adoc]: ditto as above

File: documentation/assemblies/assembly-external-logging.adoc (hunk 1 above)
Comment (PaulRMellor) [reply documentation/assemblies/assembly-external-logging.adoc]: :+1:  I changed to: the `spec.logging` property

File: documentation/assemblies/assembly-external-logging.adoc (hunk 2 above)
Comment (PaulRMellor) [reply documentation/assemblies/assembly-external-logging.adoc]: I've changed if you think it helps understanding

File: documentation/modules/proc-creating-configmap.adoc (hunk 3 above)
Comment (PaulRMellor) [reply documentation/modules/proc-creating-configmap.adoc]: Yes. We should make it obvious, so changed to: 

* `log4j.properties` for Kafka components and the Kafka Bridge

File: documentation/modules/proc-creating-configmap.adoc (hunk 4 above)
Comment (PaulRMellor) [reply documentation/modules/proc-creating-configmap.adoc]: Nice idea! Thanks

File: documentation/modules/proc-creating-configmap.adoc [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
++
+[source,yaml,subs="+attributes"]
+----
+kind: ConfigMap
+apiVersion: {KafkaApiVersion}
+metadata:
+  name: logging-configmap
+data:
+  log4j.properties:
+  # logger settings for resource
```
Comment (scholzj) [inline documentation/modules/proc-creating-configmap.adoc]: I think this comment is a bit confusing. Is it part of the log4j.properties file in which case it is misaligned? Or is it part of the YAML?

File: documentation/modules/proc-creating-configmap.adoc [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+// assembly-external-logging.adoc
+
+[id='creating-configmap_{context}']
+= Creating a ConfigMap for logging
+
+To use a ConfigMap to define logging properties, you create the ConfigMap and then reference it as part of the logging definition in the `spec` of a resource.
+
+The ConfigMap must contain the appropriate logging configuration.
+
+* `log4j.properties` for Kafka components and the Kafka Bridge
```
Comment (scholzj) [inline documentation/modules/proc-creating-configmap.adoc]: Maybe we should specifically mention Zookeeper as well?

File: documentation/modules/proc-creating-configmap.adoc [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+Here we demonstrate how a ConfigMap defines a root logger for a Kafka resource.
+
+.Procedure
+
+. Create the ConfigMap.
++
+You can create the ConfigMap as a YAML file or from a properties file at the `kubectl` command line.
++
+ConfgMap example with a root logger definition for Kafka:
```
Comment (laidan6000) [inline documentation/modules/proc-creating-configmap.adoc]: Typo: ConfigMap

File: documentation/modules/proc-creating-configmap.adoc (hunk 7 above)
Comment (PaulRMellor) [reply documentation/modules/proc-creating-configmap.adoc]: I removed it.

File: documentation/modules/proc-creating-configmap.adoc (hunk 8 above)
Comment (PaulRMellor) [reply documentation/modules/proc-creating-configmap.adoc]: Added.

File: documentation/modules/proc-creating-configmap.adoc (hunk 9 above)
Comment (PaulRMellor) [reply documentation/modules/proc-creating-configmap.adoc]: Thanks!

**Comment by PaulRMellor** [comment]
@scholzj - do we want to merge this one?

File: documentation/modules/proc-creating-configmap.adoc [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+The configuration must be placed under these properties.
+
+Here we demonstrate how a ConfigMap defines a root logger for a Kafka resource.
+
+.Procedure
+
+. Create the ConfigMap.
++
+You can create the ConfigMap as a YAML file or from a properties file at the `kubectl` command line.
```
Comment (tombentley) [inline documentation/modules/proc-creating-configmap.adoc]:
```suggestion
You can create the ConfigMap as a YAML file or from a properties file using `kubectl` at the command line.
```

**Pull request merged**